repos/fdbt-site/tests/pages/api/additionalPricingStructures.test.ts

```diff
@@ -0,0 +1,196 @@
+import { getMockRequestAndResponse } from '../../testData/mockData';
+import * as sessions from '../../../src/utils/sessions';
+import additionalPricingStructures, {
+    validateAdditionalStructuresInput,
+} from '../../../src/pages/api/additionalPricingStructures';
+import { ErrorInfo, AdditionalPricing } from '../../../src/interfaces';
+import { ADDITIONAL_PRICING_ATTRIBUTE } from '../../../src/constants/attributes';
+
+describe('additionalPricingStructures', () => {
+    const updateSessionAttributeSpy = jest.spyOn(sessions, 'updateSessionAttribute');
+    const writeHeadMock = jest.fn();
+
+    afterEach(() => {
+        jest.resetAllMocks();
+    });
+
+    it('correctly generates additional pricing structures info, updates the  ADDITIONAL_PRICING_ATTRIBUTE and then redirects to /capConfirmation if all is valid', () => {
+        const mockAdditionalStructuresInfo: AdditionalPricing = {
+            pricingStructureStart: '2',
+            structureDiscount: '2',
+        };
+
+        const { req, res } = getMockRequestAndResponse({
+            body: {
+                additionalDiscounts: 'yes',
+                pricingStructureStart: '2',
```

Comment:
I think we need a test to make sure we are NOT setting the ADDITIONAL_PRICING_ATTRIBUTE if the user selects no.